lib/helper.js

```diff
@@ -48,7 +50,53 @@ function parseTemplateConfigName(config) {
     return {};
 }
 
+/**
+ * recursively fetching records (ex: builds/events) until the end
+ * @method getAllRecords
+ * @param  {String}       funcName    Function to fetch. For example: 'getBuilds'
+ * @param  {opts}         opts        opts to pass into scan function
+ * @param  {Array}        resultArray Result so far
+ * @param  {String}       date        Current date for aggregation
+ * @param  {Number}       index       Current index (each index is a different date)
+ * @return {Array}                    Array where each element is an array of records belong to the same date
+ */
+async function getAllRecords(funcName, opts, resultArray, date, index) {
+    const builds = await this[funcName](opts);
+    const length = builds.length;
```

Comment:
do you need make sure `builds` is an array?

Reply:
it's from `list` methods so should always be an array?